dec/decode.go

```diff
@@ -77,14 +77,20 @@ type BrotliReader struct {
 	// Internal buffer for compressed data
 	buffer []byte
 
-	availableIn C.size_t
-	nextIn      *C.uint8_t
-	totalOut    C.size_t
+	availableIn     C.size_t
+	nextIn          *C.uint8_t
+	totalOut        C.size_t
+	eof             bool
```

Comment:
I find it dangerous that this library doesn't make errors persistent. That is, if I encounter a brotli error, calling Read again should give me the same error.

What you could do is store ```err error``` and set it to io.EOF when the stream ends successfully, otherwise set ```err``` to the error that was raised.